File: rgctl-tests/ecommerce-go/internal/models/category.go
package models

type Category struct {
	ID        string `gorm:"primaryKey" json:"id"`
	Name      string `gorm:"not null" json:"name"`
	Slug      string `gorm:"uniqueIndex;not null" json:"slug"`
	CreatedAt string `gorm:"not null" json:"created_at"`
}

func (c *Category) SetCreatedAt(v string) { c.CreatedAt = v }

File: rgctl-tests/ecommerce-python/app/models/category.py
from sqlalchemy import String
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base
from app.models.base import TimestampMixin


class Category(Base, TimestampMixin):
    __tablename__ = "categories"

    id: Mapped[str] = mapped_column(String(36), primary_key=True)
    name: Mapped[str] = mapped_column(String(255))
    slug: Mapped[str] = mapped_column(String(255), unique=True, index=True)

    products = relationship("Product", back_populates="category", cascade="all, delete-orphan")
